M2/Macaulay2/m2/testing.m2

```diff
@@ -35,13 +23,21 @@ editMethod TestInput := editMethod @@ locate
 TEST = method(Options => {FileName => false})
 TEST List   := opts -> testlist   -> apply(testlist, test -> TEST(test, opts))
 TEST String := opts -> teststring -> (
-    n := currentPackage#"test number";
-    currentPackage#"test inputs"#n = TestInput if opts.FileName then (
-        testCode := get teststring;
-        (minimizeFilename teststring, depth net testCode + 1, testCode)
-        ) else
-        (minimizeFilename currentFileName, currentRowNumber(), teststring);
-    currentPackage#"test number" = n + 1;)
+    n := #currentPackage#"test inputs";
+    if opts.FileName then (
```

Comment:
Is this option ever used? I have seen `TEST get "test.m2"` but never this option. Wouldn't be opposed to deprecating it for simplicity.

Reply:
Oooh actually this seems like a bug:
```m2
i1 : TEST get "../../Macaulay2/packages/Macaulay2Doc/demo1.m2"

i2 : tests User

o2 = {0 => TestInput[stdio:-13:1-1:80]}

i3 : locate oo

o3 = {0 => (stdio:-13:1-1:80)}

i4 : code oo

o4 = stdio:-13:1-1:80: --source code:














     -- This is the beginning of your Macaulay2 log stored at /home/mahrud/.Macaulay2/history.m2
```
Sorry to be making this PR more complicated for you!

Reply:
The `FileName` option is used for all the `Core` tests.  `JSON` too.  I'm not sure if there are any others -- it's relatively new.

Reply:
Hmm, yeah that's a tricky bug.  I think it's probably always existed for `locate`, since it assumes we're calling `TEST` on a literal string and not the output of some function.  It's just now that `code` is calling `locate`, it's showing up there, too.

I guess the documentation for `TEST` could discourage the use of `get` for this reason, and we could modify the handful of tests that currently use it.  (I'm only counting 13.)

Reply:
Let's first try to find a way to fix the bug because I'd rather keep allowing `TEST get ...`, it's much simpler than using an option.